include/validator/context.h

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2024 Second State INC
+#include "ast/module.h"
+#include "common/errinfo.h"
+
+#include <optional>
+#include <vector>
+
+namespace WasmEdge {
+namespace Validator {
+
+class Context {
+public:
+  struct ComponentContext {
+    const AST::Component::Component *Component;
+    std::vector<AST::Module> CoreModules;
+    std::vector<AST::Component::Component> ChildComponents;
+    std::optional<const ComponentContext *> parent;
+
+    ComponentContext(const AST::Component::Component *C,
+                     std::optional<const ComponentContext *> p = std::nullopt)
+        : Component(C), CoreModules(), ChildComponents(), parent(p) {}
+  };
+
+  void enterComponent(const AST::Component::Component &C) {
+    if (!ComponentContexts.empty()) {
+      ComponentContexts.back().ChildComponents.emplace_back(C);
+    }
+    std::optional<const ComponentContext *> parent =
+        ComponentContexts.empty() ? std::nullopt
+                                  : std::optional{&ComponentContexts.back()};
+    ComponentContexts.emplace_back(&C, parent);
+  }
+
+  void exitComponent() {
+    if (!ComponentContexts.empty()) {
+      ComponentContexts.pop_back();
+    } else {
+      assumingUnreachable();
+    }
+  }
+
+  const AST::Component::Component *getCurrentComponent() const {
+    return ComponentContexts.empty() ? nullptr
+                                     : ComponentContexts.back().Component;
+  }
```

Comment:
Where is this method got invoked? It would be more proper if we just make **the** current component as its argument.